include/rocksdb/utilities/customizable_util.h

```diff
@@ -0,0 +1,293 @@
+  } else {
+#ifndef ROCKSDB_LITE
+    status = config_options.registry->NewUniqueObject(id, result);
+#else
+    status = Status::NotSupported("Cannot load object in LITE mode ", id);
+#endif  // ROCKSDB_LITE
+    if (config_options.ignore_unsupported_options && status.IsNotSupported()) {
+      return Status::OK();
+    }
+  }
+  if (!status.ok() || opt_map.empty()) {
+    return status;
+  } else if (result->get() != nullptr) {
+    return result->get()->ConfigureFromMap(config_options, opt_map);
+  } else {
+    return Status::InvalidArgument("Cannot configure null object ");
+  }
+}
+
+// Creates a new unique customizable instance object based on the input
+// parameters.
+// @see LoadSharedObject for more information on the inner workings of this
+// method.
+//
+// @param config_options Controls how the instance is created and errors are
+// handled
+// @param value Either the simple name of the instance to create, or a set of
+// name-value pairs to create and initailize the object
+// @param func  Optional function to call to attempt to create an instance
+// @param result The newly created instance.
+template <typename T>
+static Status LoadUniqueObject(const ConfigOptions& config_options,
+                               const std::string& value,
+                               const UniqueFactoryFunc<T>& func,
+                               std::unique_ptr<T>* result) {
+  std::string id;
+  std::unordered_map<std::string, std::string> opt_map;
+  Status status = Customizable::GetOptionsMap(config_options, result->get(),
+                                              value, &id, &opt_map);
+  if (!status.ok()) {  // GetOptionsMap failed
+    return status;
+  } else if (func == nullptr ||
+             !func(id, result)) {  // No factory, or it failed
+    if (value.empty()) {           // No Id and no options.  Clear the object
+      *result = nullptr;
+      return Status::OK();
+    } else {
+      return NewUniqueObject(config_options, id, opt_map, result);
+    }
+  } else if (opt_map.empty()) {
+    return status;
+  } else if (result->get() != nullptr) {
+    return result->get()->ConfigureFromMap(config_options, opt_map);
+  } else {
+    return Status::InvalidArgument("Cannot configure null object ");
+  }
+}
+
+// Creates a new static (raw pointer) customizable instance object based on the
+// input parameters using the object registry.
+// @see NewSharedObject for more information on the inner workings of this
+// method.
+//
+// @param config_options Controls how the instance is created and errors are
+// handled
+// @param id The identifier of the new object being created.  This string
+// will be used by the object registry to locate the appropriate object to
+// create.
+// @param opt_map Optional name-value pairs of properties to set for the newly
+// created object
+// @param result The newly created and configured instance.
+template <typename T>
+static Status NewStaticObject(
```

Comment:
Just a general question, can we replace some of the current static object function with this interface? For example, when we new a default env, FS, rate-limiter?

Reply:
I think the CreateFromString methods can eventually many of the extern constructors.  The methods in this file are meant more for use by developers of classes rather than users